packages/core/core/src/services/document-service/repository.ts

```diff
@@ -278,28 +287,35 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
       i18n.multiLocaleToLookup(contentType)
     )(params);
 
-    // Delete all drafts that match query
-    await deleteDocument({
-      ...queryParams,
-      documentId,
-      lookup: { ...queryParams?.lookup, publishedAt: null },
-    });
-
-    // Get deep populate of published versions
-    const entriesToDraft = await strapi.db?.query(uid).findMany({
-      where: {
-        ...queryParams?.lookup,
-        documentId,
-        publishedAt: { $ne: null },
-      },
-      populate: getDeepPopulate(uid, { relationalFields: ['documentId', 'locale'] }),
-    });
+    const [versionsToDraft, versionsToDelete] = await Promise.all([
+      strapi.db?.query(uid).findMany({
```

Comment:
Can query be undefined on db? I would find that a little confusing. If it can be then I guess you may end up with undefined instead of an array. It looks like findMany can also return undefined. If that's the case couldn't that be an issue if you are iterating over an expected array value?

Reply:
indeed it can not be undefined! will change that :)